src/main/java/com/gtnewhorizon/gtnhlib/util/data/BlockMeta.java

```diff
@@ -0,0 +1,79 @@
+package com.gtnewhorizon.gtnhlib.util.data;
+
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+
+import net.minecraft.block.Block;
+
+/**
+ * A mutable implementation of {@link ImmutableBlockMeta}. If your API should return a mutable pair, return this
+ * instead. Must follow the same contracts as the immutable version if this is ever upcast to a
+ * {@link ImmutableBlockMeta} in your API. If this type is exposed instead of the immutable interface, assume that the
+ * contained values can change.
+ */
+public class BlockMeta implements ImmutableBlockMeta {
+
+    @Nonnull
+    private Block block;
+    private int meta;
+
+    public BlockMeta(Block block, int meta) {
+        this.block = block;
```

Comment:
The constructors would need to have the same nullability treatment as the setters. If performance is a concern it's probably fine to rely on the annotation and drop the `Objects.requireNonNull` call, though it can't be that expensive.